oneoffs/joseki/joseki_query.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import sys
+sys.path.insert(0, '.')  # nopep8
+import logging
+import json
+import sqlite3
+import collections
+import datetime as dt
+
+from flask import Flask, g
+from timeit import default_timer as timer
+
+import os
+import flask
+
+import oneoffs.joseki.opening_freqs as openings
+
+# Suppress Flask's info logging.
+log = logging.getLogger("werkzeug")
+log.setLevel(logging.WARNING)
+
+# static_folder is location of npm build
+app = Flask(__name__, static_url_path="", static_folder="./build")
+
+DATABASE = '/data/sgf/wr_joseki_3000.db' # relative to joseki_query.py
+
+def get_db():
+    db = getattr(g, '_database', None)
+    if db is None:
+        db = g._database = sqlite3.connect(DATABASE)
+    return db
+
+@app.teardown_appcontext
+def close_connection(exception):
+    db = getattr(g, '_database', None)
+    if db is not None:
+        db.close()
+
+
+def query_db(seq):
+    cur = get_db().execute('''
+                     select run, hour, count from joseki_counts
+                           where seq_id = (select id from joseki where seq = ?);
+                     ''', (seq, ))
+    return list(cur.fetchall())
+
+@app.route("/")
+def index():
+    return app.send_static_file("index.html")
+
+@app.route("/nexts", methods=["POST"])
+def nexts():
+    d = json.loads(flask.request.data.decode('utf-8'))
+    prefix = d['params']['prefix']
+    run = d['params']['run']
+
+    db = get_db()
+
+    res = {'count': 0, 'next_moves': {}}
+
+    if not prefix:
+        nexts = db.execute("select distinct(substr(seq, 0, 7)), count(*) from joseki group  by 1;").fetchall()
+        total = sum([n[1] for n in nexts])
+    else:
+        s_id = db.execute("select id from joseki where seq=?", (prefix,)).fetchone()
+        if not s_id:
+            return flask.Response(json.dumps(res), mimetype='text/json')
```

Comment:
consider:

from flask import jsonify
return jsonify(res)

this handles setting status=200 and mimetype